Add Dockerfile and improve README (#133)
SUSE/pcw: README.md

# OpenQA Public cloud Helper

> Publiccloud-Watcher or pcw is a a web app which monitors, displays and deletes resources on various cloud service providers (CSPs) and linked resources. It identifies abandoned instances by searching for certain tags and performs cleanup tasks. It also deletes old custom uploaded images. Behavior differs per CSP.

Reviewer: ```suggestion
PublicCloud-Watcher or pcw is a web app which monitors, displays and deletes resources on various Cloud Service Providers (CSPs). It identifies abandoned instances by searching for certain tags and performs cleanup tasks. It also deletes old custom uploaded images. The behavior differs per CSP.
```
not sure what you mean about `... and linked resources`

Author: Yeah, that's too vague, especially for the first paragraph. I removed it as you suggested, that reads nice.

Configuration of Publiccloud-Watcher happens via a global config file in `/etc/pcw.ini`. See [templates/pcw.ini](templates/pcw.ini) for a configuration template. To start, copy the template over:

    cp templates/pwc.ini /etc/pcw.ini

The bare minimum configuration requires the `vault[user]` and `vault[password]` settings.

PCW supports email notifications about left-over instances. See the `notify` section therein and their corresponding comments.

```bash
# Setup virtual environment
virtualenv env
source env/bin/activate
pip install -r requirements.txt

## Configuration steps, only required once to setup the database and user
# Setup database
python manage.py migrate
# Setup superuser
python manage.py createsuperuser --email admin@example.com --username admin
python manage.py collectstatic

## Running the webapp server
python manage.py runserver
```

By default, PCW runs on http://127.0.0.1:8000/

## Building a container

To build a docker/podman container with the default `suse/qac/pcw` tag, run

    make docker-container
    make podman-container

This repository contains the skeleton `Dockerfile` for building a Publiccloud-Watcher docker/podman container.

> The resulting container requires the `/conf` volume to be mounted and contain the `pcw.ini` file.

Reviewer: ```suggestion
The resulting container requires the `/conf` volume to be mounted which shall contain the `pcw.ini` file.
```

To run the resulting container using e.g. the config directory `/srv/pcw` (must contain `pcw.ini`!), the corresponding environment variables and publish port 8000, do

    podman run -ti --rm --hostname pcw --name pcw -v /srv/pcw:/conf -p 8000:8000/tcp suse/qac/pcw

> For usage in docker simply replace `podman` with `docker` in the above command.

Reviewer: ```suggestion
For usage in docker simply replace `podman` by `docker` in the above command.
```

[updaterun](ocw/management/commands/updaterun.py)

> those allows triggering core functionality without web UI. It is highly recommended to use `dry_run = True` in `pcw.ini` in

Reviewer: ```suggestion
those allow triggering core functionality without web UI. It is highly recommended to use `dry_run = True` in `pcw.ini` in
```
